code/datums/components/profound_fisher.dm

```diff
@@ -2,15 +2,18 @@
 /datum/component/profound_fisher
 	///the fishing rod this mob will use
 	var/obj/item/fishing_rod/mob_fisher/our_rod
+	///Wether we should delete the fishing rod along with the component or replace it if it's somehow removed from the parent
+	var/delete_rod = TRUE
```

Comment:
I think the variable name could be a little more descriptive about what circumstances under which we delete the rod, the comment elaborates but self-describing is better.